jib-cli/src/main/java/com/google/cloud/tools/jib/cli/jar/JarProcessors.java

```diff
@@ -65,4 +80,40 @@ private static String determineJarType(Path jarPath) throws IOException {
       return STANDARD;
     }
   }
+
+  /**
+   * Determines the java version of JAR. Derives the version from the first .class file it finds in
+   * the JAR.
+   *
+   * @param jarPath path to the jar
+   * @return java version
+   * @throws IOException if I/O exception thrown when opening the jar file
+   */
+  private static Integer getVersion(Path jarPath) throws IOException {
+    try (JarFile jarFile = new JarFile(jarPath.toFile())) {
+      Enumeration<JarEntry> jarEntries = jarFile.entries();
+      while (jarEntries.hasMoreElements()) {
+        String jarEntry = jarEntries.nextElement().toString();
+        if (jarEntry.endsWith(".class")) {
+          URLClassLoader loader = new URLClassLoader(new URL[] {jarPath.toUri().toURL()});
+          try (InputStream classFileInfo = loader.getResourceAsStream(jarEntry)) {
+            DataInputStream data = new DataInputStream(classFileInfo);
+
+            // Check magic number
+            if (data.readInt() != 0xCAFEBABE) {
+              throw new IOException("Invalid class file format.");
```

Comment:
I guess this should work with the Java 9 [`module-info.class`](https://stackoverflow.com/q/57358750/1701388), but it's worth testing.

I found a [simple example](https://github.com/tfij/Java-9-modules---the-simplest-example), where running `compile.sh` generates `build/pl.tfij.java9modules.app/module-info.class` and running `package.sh` generates `mods/*.jar`. I'd also be interesting to see if `jib jar pl.tfij.java9modules.app@1.0.jar` works fine.

Reply:
Oh nice. yes, it is worth double-checking. I tested it with the example you suggested above and can confirm that the version calculated is indeed 9.
A snippet of the test:
```
  Path jarPath = Paths.get(Resources.getResource(JAVA_9_JAR).toURI());
  Integer version = JarProcessors.getVersion(jarPath);
  assertThat(version).isEqualTo(9);
```

Reply:
Oh, I should have been clearer. Can you test this section of code on the compiled `module-info.class`? If you use a jar, it's possible that the code retrieved the version from a different `.class` file. I'm trying to see if the `module-info.class` uses the same magic number and the structure.

Also, have you tested `jib jar pl.tfij.java9modules.app@1.0.jar`?

Reply:
BTW, it doesn't have to be compiled with Java 9. If you use Java 11, it should report 11.

Reply:
Ahh I see, sorry. I tested this again and there's something interesting happening here. This method returns 9 for `module-info.class` but 11 (which is what I used while running `compile.sh` and `package.sh`) for `pl.tfij.java9modules.app.ModuleApp.class`. I need to look into this more but I'm wondering if `module-info.class` is always in java 9?

> Also, have you tested `jib jar pl.tfij.java9modules.app@1.0.jar`?

And running this results in a ` java.lang.ClassNotFoundException: pl.tfij.java9modules.greetings.Greeting`. In terms of how our code is set up, since that jar's manifest doesn't specify a Class-path, there isn't a way for us to determine what the dependency is.

Reply:
Ah, that makes sense. I think we should skip `module-info.class`. Not sure why, but it looks like the number is always Java 9.

> In terms of how our code is set up, since that jar's manifest doesn't specify a `Class-path`, there isn't a way for us to determine what the dependency is.

Aha. I see that `run.sh` doesn't run the jar with `java -jar` but with `java --module-path ...` (which corresponds to `java --class-path` in Java <=8 in some sense). I tried to do a quick search, but I'm not sure how one can build a runnable jar with modules that require dependency jars. We should revisit the Java 9+ module at some point. For this PR, let's just focus on determining the correct Java version.